### ISSUE TITLE
DWIConvert alters bval/bvector values

### ISSUE THREAD

**Issue by pmolfese** [BRAINSia/BRAINSTools#183]
I've noticed that DWIConvert gets different b-values and b-vectors than what is represented in the DICOM files either when inspecting the DICOM files or using dcm2nii to grab values.  For example, with a b-value of 1000, DWIConvert seems to interpret this as 1002.  

And while the first gradient out of dcm2nii is reported as:
 0.0471797 -0.906431 0.419709

The corresponding vector from DWIConvert is read as:
-0.047709 -0.920511 -0.387794

Is there some reorientation of the data going on during the conversion?  The dcm2nii values match what we have coded into the scanner.

**Comment by hjmjohnson** [comment]
I thought we addressed this issue already in November of last year:  https://github.com/BRAINSia/BRAINSTools/issues/159

**Comment by hjmjohnson** [comment]
George,

Does the DWI flag

   --transposeInputBVectors
     FSL input BVectors are expected to be encoded in the input file as one
     vector per line. If it is not the case, use this option to transpose
     the file as it is read (default: 0)

fix your issues?

This was added last year:
commit 7598454f22ade7d2b5b30b5706749f20e93ccbb7
Author: Hans Johnson <hans-johnson@uiowa.edu>
Date:   Wed Nov 26 16:08:03 2014 -0600

    COMP: Incorrect function signature.
    
    After adding transpose option to DWIConvert, the
    ReadBVecs function signature was incorrect.


I am wondering why that feature (https://github.com/BRAINSia/BRAINSTools/issues/159) is not working.

Hans

**Comment by hjmjohnson** [comment]
https://github.com/BRAINSia/BRAINSTools/commit/4dbc02d75530934cd5370aff7fdcf1fd893af069
https://github.com/BRAINSia/BRAINSTools/commit/b2fd801d6a7d1a99048de3a4dcc66b6c26e008e6

Fixed in these two patches (along with other fixes).

**Comment by dtosun** [comment]
I have a similar issue. 

I just installed the latest Slicer release (4.5.0-1 for macosx). 

If I run DWIConvert from the Slicer user interface to convert an nhdr/raw.gz image to a nifti image, it creates the bvec and bval files with correct entries but doesn't create a .nii image file.

Then if I use the command line to convert the nhdr/raw.gz image to a nii image, it creates the nii file properly but this time the values in the bvec and bval are totally off and looks totally random.

Here is the command I used:

Slicer.app/Contents/lib/Slicer-4.5/cli-modules/DWIConvert \
--inputVolume /path/to/input/Subject1.nhdr \
-o /path/to/output/Subject1.nii \
--outputBVectors /path/to/output/Subject1.bvec \
--outputBValues /path/to/output/Subject1.bval \
--conversionMode NrrdToFSL

Is there any particular flag or switch I should have used with the command line version?
Thanks
